### PR TITLE
Fix SecurityException when opening files

### LINKED ISSUE #2223: File type processed as NFC
## Exception 
* __User Action:__ UI Error 
* __Request:__ Application crash 
* __Version:__ 3.5.3 
* __OS:__ Linux Android 10 - 29 
* __Device:__ a71 
* __Model:__ SM-A715W 
* __Product:__ a71cs 

Crash log

``` java.lang.SecurityException: Permission Denial: starting Intent { act=android.intent.action.VIEW dat=content://com.amaze.filemanager/storage_root/storage/emulated/0/Download/journal typ=*/* flg=0x1 cmp=com.android.apps.tag/.TagViewer } from ProcessRecord{e88b5a3 1222:com.amaze.filemanager/u0a522} (pid=1222, uid=10522) requires android.permission.NFC at android.os.Parcel.createException(Parcel.java:2088) at android.os.Parcel.readException(Parcel.java:2056) at android.os.Parcel.readException(Parcel.java:2004) at android.app.IActivityTaskManager$Stub$Proxy.startActivity(IActivityTaskManager.java:4454) at android.app.Instrumentation.execStartActivity(Instrumentation.java:1716) at android.app.Activity.startActivityForResult(Activity.java:5258) at androidx.fragment.app.FragmentActivity.startActivityForResult(FragmentActivity.java:676) at android.app.Activity.startActivityForResult(Activity.java:5203) at androidx.fragment.app.FragmentActivity.startActivityForResult(FragmentActivity.java:663) at android.app.Activity.startActivity(Activity.java:5587) at android.app.Activity.startActivity(Activity.java:5555) at com.amaze.filemanager.ui.dialogs.OpenFileDialogFragment$loadViews$$inlined$let$lambda$1.onClick(OpenFileDialogFragment.kt:344) at android.view.View.performClick(View.java:7862) at android.widget.TextView.performClick(TextView.java:15004) at android.view.View.performClickInternal(View.java:7831) at android.view.View.access$3600(View.java:879) at android.view.View$PerformClick.run(View.java:29359) at android.os.Handler.handleCallback(Handler.java:883) at android.os.Handler.dispatchMessage(Handler.java:100) at android.os.Looper.loop(Looper.java:237) at android.app.ActivityThread.main(ActivityThread.java:8167) at java.lang.reflect.Method.invoke(Native Method) at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496) at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100) Caused by: android.os.RemoteException: Remote stack trace: at com.android.server.wm.ActivityStackSupervisor.checkStartAnyActivityPermission(ActivityStackSupervisor.java:1447) at com.android.server.wm.ActivityStarter.startActivity(ActivityStarter.java:980) at com.android.server.wm.ActivityStarter.startActivity(ActivityStarter.java:732) at com.android.server.wm.ActivityStarter.startActivityMayWait(ActivityStarter.java:2074) at com.android.server.wm.ActivityStarter.execute(ActivityStarter.java:646) ```

Get Outlook for Android<https://aka.ms/ghei36>

### LINKED ISSUE #2329: SecurityException Intent requires CALL_PHONE
## Exception
* __User Action:__ UI Error
* __Request:__ Application crash
* __Version:__ 3.5.3
* __OS:__ Linux
samsung/c1qsqw/c1q:11/RP1A.200720.012/N981USQU1CTLA:user/release-keys 11 -
30
* __Device:__ c1q
* __Model:__ SM-N981U
* __Product:__ c1qsqw
<details><summary><b>Crash log </b></summary><p>

```
java.lang.SecurityException: Permission Denial: starting Intent {
act=android.intent.action.VIEW
dat=content://com.amaze.filemanager/storage_root/storage/emulated/0/.chartboost/html/mraid-iframe-open-414071600
typ=*/* flg=0x1
cmp=com.google.android.gms/.matchstick.call.ContactsVideoActionActivity }
from ProcessRecord{5bd4c83 4965:com.amaze.filemanager/u0a303} (pid=4965,
uid=10303) requires android.permission.CALL_PHONE
at android.os.Parcel.createExceptionOrNull(Parcel.java:2385)
at android.os.Parcel.createException(Parcel.java:2369)
at android.os.Parcel.readException(Parcel.java:2352)
at android.os.Parcel.readException(Parcel.java:2294)
at
android.app.IActivityTaskManager$Stub$Proxy.startActivity(IActivityTaskManager.java:4225)
at android.app.Instrumentation.execStartActivity(Instrumentation.java:1723)
at android.app.Activity.startActivityForResult(Activity.java:5374)
at
androidx.fragment.app.FragmentActivity.startActivityForResult(FragmentActivity.java:676)
at android.app.Activity.startActivityForResult(Activity.java:5332)
at
androidx.fragment.app.FragmentActivity.startActivityForResult(FragmentActivity.java:663)
at android.app.Activity.startActivity(Activity.java:5718)
at android.app.Activity.startActivity(Activity.java:5671)
at
com.amaze.filemanager.adapters.AppsAdapter.startActivityForRowItem(AppsAdapter.java:192)
at
com.amaze.filemanager.adapters.AppsAdapter.lambda$getView$0(AppsAdapter.java:166)
at
com.amaze.filemanager.adapters.AppsAdapter.lambda$getView$0$AppsAdapter(Unknown
Source:0)
at
com.amaze.filemanager.adapters.-$$Lambda$AppsAdapter$O2jRE_NHwDMQ5NBArFexYNoqVHk.onClick(Unknown
Source:4)
at android.view.View.performClick(View.java:8178)
at android.view.View.performClickInternal(View.java:8147)
at android.view.View.access$3700(View.java:888)
at android.view.View$PerformClick.run(View.java:30233)
at android.os.Handler.handleCallback(Handler.java:938)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:246)
at android.app.ActivityThread.main(ActivityThread.java:8429)
at java.lang.reflect.Method.invoke(Native Method)
at
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:596)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
Caused by: android.os.RemoteException: Remote stack trace:
at
com.android.server.wm.ActivityStackSupervisor.checkStartAnyActivityPermission(ActivityStackSupervisor.java:1315)
at
com.android.server.wm.ActivityStarter.executeRequest(ActivityStarter.java:1242)
at com.android.server.wm.ActivityStarter.execute(ActivityStarter.java:876)
at
com.android.server.wm.ActivityTaskManagerService.startActivityAsUser(ActivityTaskManagerService.java:1705)
at
com.android.server.wm.ActivityTaskManagerService.startActivityAsUser(ActivityTaskManagerService.java:1556)


```
</details>
<hr>

### LINKED ISSUE #2467: Crash on intent that requires call permission
## Issue explanation (write below this line)

I think I'm hacked

## Exception
* __User Action:__ UI Error
* __Request:__ Application crash
* __Version:__ 3.5.3
* __OS:__ Linux Android 9 - 28
* __Device:__ surfna
* __Model:__ moto e6 (XT2005DL)
* __Product:__ surfna_trac
<details><summary><b>Crash log </b></summary><p>

```
java.lang.SecurityException: Permission Denial: starting Intent { act=android.intent.action.VIEW dat=content://com.amaze.filemanager/storage_root/storage/emulated/0/Android/data/com.amaze.filemanager/cache/recoveredmedia/0backup_20210402_205158.cmb typ=*/* flg=0x1 cmp=com.google.android.apps.tachyon/.ContactsAudioActionActivity2 } from ProcessRecord{d1c76b0 4653:com.amaze.filemanager/u0a429} (pid=4653, uid=10429) requires android.permission.CALL_PRIVILEGED
	at android.os.Parcel.createException(Parcel.java:1950)
	at android.os.Parcel.readException(Parcel.java:1918)
	at android.os.Parcel.readException(Parcel.java:1868)
	at android.app.IActivityManager$Stub$Proxy.startActivity(IActivityManager.java:3692)
	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1738)
	at android.app.Activity.startActivityForResult(Activity.java:4594)
	at androidx.fragment.app.FragmentActivity.startActivityForResult(FragmentActivity.java:676)
	at android.app.Activity.startActivityForResult(Activity.java:4552)
	at androidx.fragment.app.FragmentActivity.startActivityForResult(FragmentActivity.java:663)
	at android.app.Activity.startActivity(Activity.java:4917)
	at android.app.Activity.startActivity(Activity.java:4885)
	at com.amaze.filemanager.adapters.AppsAdapter.startActivityForRowItem(AppsAdapter.java:192)
	at com.amaze.filemanager.adapters.AppsAdapter.lambda$getView$0(AppsAdapter.java:166)
	at com.amaze.filemanager.adapters.AppsAdapter.lambda$getView$0$AppsAdapter(Unknown Source:0)
	at com.amaze.filemanager.adapters.-$$Lambda$AppsAdapter$O2jRE_NHwDMQ5NBArFexYNoqVHk.onClick(Unknown Source:4)
	at android.view.View.performClick(View.java:6600)
	at android.view.View.performClickInternal(View.java:6577)
	at android.view.View.access$3100(View.java:781)
	at android.view.View$PerformClick.run(View.java:25917)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:215)
	at android.app.ActivityThread.main(ActivityThread.java:6939)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:870)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.am.ActivityStackSupervisor.checkStartAnyActivityPermission(ActivityStackSupervisor.java:1892)
	at com.android.server.am.ActivityStarter.startActivity(ActivityStarter.java:735)
	at com.android.server.am.ActivityStarter.startActivity(ActivityStarter.java:559)
	at com.android.server.am.ActivityStarter.startActivityMayWait(ActivityStarter.java:1139)
	at com.android.server.am.ActivityStarter.execute(ActivityStarter.java:500)


```
</details>
<hr>

Sent from Yahoo Mail on Android

### PR DESCRIPTION
**Pull request by EmmanuelMess**
## Description

Simple catch for `SecurityException` so that it doesn't crash in the user's face. 

#### Issue tracker   

Fixes #2223
Fixes #2329
Fixes #2467

#### Automatic tests

- [ ] Added test cases
  
#### Manual tests
- [ ] Done  
  

#### Build tasks success  

Successfully running following tasks on local:
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

### CODE REVIEW

File: app/src/main/java/com/amaze/filemanager/ui/dialogs/OpenFileDialogFragment.kt [hunk 1]
```diff
@@ -385,6 +387,15 @@ class OpenFileDialogFragment : BaseBottomSheetFragment() {
         }
     }
 
+    private fun startActivityCatchingNFC(intent: Intent) {
```
Comment (TranceLove) [inline app/src/main/java/com/amaze/filemanager/ui/dialogs/OpenFileDialogFragment.kt:390]: I think there are more `Activity`s that would trigger `SecurityException`, like #2329. Better use a more generic name for this.

**Comment by TranceLove** [comment]
Actually I thought of blacklisting `Activity`s like NFC or Phone app to prevent this exception though. But anyway...

**Comment by EmmanuelMess** [comment]
> Actually I thought of blacklisting `Activity`s like NFC or Phone app to prevent this exception though. But anyway...

How would you say of doing that?

**Comment by EmmanuelMess** [comment]
I was under the impression that these exceptions occur only because the user is trying to open a file in an incorrect app (like a phone app or an NFC reader app for a text file).

**Review by VishalNehra — DISMISSED** [review]

**Comment by TranceLove** [comment]
> > Actually I thought of blacklisting `Activity`s like NFC or Phone app to prevent this exception though. But anyway...
> 
> How would you say of doing that?

Sometimes it's better to prevent errors by forbidding them from doing wrong actions.

But to reiterate, I'm OK with this approach too... just wonder what files/file content can be opened successfully with a NFC or Phone app.

**Review by TranceLove — APPROVED** [review]
Tested OK on Pixel 2 emulator running Android 11, with the cases of tapping "Just Once", "Always" or choosing the app directly from the list, exceptions were successfully captured and Toasted user.

+1.

**Review by VishalNehra — APPROVED** [review]

**Pull request merged**